__tests__/emotion.js

```diff
@@ -0,0 +1,288 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import createCache from '@emotion/cache'
+import { CacheProvider } from '@emotion/react'
+import styled from '@emotion/styled'
+import { render } from '@testing-library/react'
+import prettier from 'prettier'
+import React, { useState } from 'react'
+import stylisExtraScope from '..'
+
+const formatCss = (css) => prettier.format(css, { parser: 'css' })
+const getAllEmotionStyles = () =>
+  formatCss(
+    [].slice
+      .call(document.querySelectorAll('[data-emotion]'))
+      .map((style) => style.textContent)
+      .join('\n'),
+  )
+
+const TestCacheProvider = ({ children }) => {
+  const [emotionCache] = useState(() => {
+    return createCache({
+      key: 'extra-scope',
+      stylisPlugins: [stylisExtraScope('#extraScopeContainer.extraScope')],
+    })
+  })
+
+  return <CacheProvider value={emotionCache}>{children}</CacheProvider>
+}
+
+describe('StylisExtraScope with Emotion', () => {
+  beforeEach(() => {
+    document.querySelector('head').innerHTML = ''
+    document.querySelector('body').innerHTML = ''
+  })
+
+  it('should handle simple input', () => {
+    const Comp1 = styled.div({ fontSize: '10px' })
+
+    render(
+      <TestCacheProvider>
+        <Comp1 />
+      </TestCacheProvider>,
+    )
+
+    expect(getAllEmotionStyles()).toMatchInlineSnapshot(`
```

Comment:
@drenckpohl I've pushed out some changes to use inline snapshots for this - it's much easier to review this if I don't have to jump between files 😉